core/lib/zksync_core/src/consensus/storage/mod.rs

```diff
@@ -259,14 +280,47 @@ impl PersistentBlockStore for BlockStore {
             .context("genesis is missing")?)
     }
 
-    async fn last(&self, ctx: &ctx::Ctx) -> ctx::Result<Option<validator::CommitQC>> {
-        self.inner
-            .access(ctx)
+    async fn state(&self, ctx: &ctx::Ctx) -> ctx::Result<BlockStoreState> {
+        let mut conn = self.inner.access(ctx).await.wrap("access()")?;
+
+        // Fetch the range of miniblocks in storage.
+        let block_range = conn.block_range(ctx).await.context("block_range")?;
+
+        // Fetch the range of certificates in storage.
+        let genesis = conn
+            .genesis(ctx)
             .await
-            .wrap("access()")?
+            .wrap("genesis()")?
+            .context("genesis missing")?;
+        let first_cert = genesis.fork.first_block.max(block_range.start);
```

Comment:
Bikeshedding: IMO, having a block number variable named `*_cert` is misleading. Also, it could make sense to call this something like `first_expected_cert` (or `number_of_first_expected_cert`?) to contrast it with `last_cert`, which it the *actual* last certificate in the storage.

Reply:
done